map.ipynb

```diff
@@ -0,0 +1,386 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "import shapely.geometry\n",
+    "import requests\n",
+    "import netCDF4\n",
+    "import tempfile\n",
+    "import json\n",
+    "import time\n",
+    "\n",
+    "from ipywidgets import *\n",
+    "from ipyleaflet import *"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "f = open('domains.json')\n",
+    "data = json.load(f)\n",
+    "\n",
+    "peace_coords = [\n",
+    "    (coord[0], coord[1]) for coord in data['borders']['peace']['coordinates']\n",
+    "]\n",
+    "fraser_coords = [\n",
+    "    (coord[0], coord[1]) for coord in data['borders']['fraser']['coordinates']\n",
+    "]\n",
+    "columbia_coords = [\n",
+    "    (coord[0], coord[1]) for coord in data['borders']['columbia']['coordinates']\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "def handle_click(**kwargs):\n",
+    "    if kwargs.get('type') == 'click':\n",
+    "        lat = round(kwargs.get('coordinates')[0], 5)\n",
+    "        lon = round(kwargs.get('coordinates')[1], 5)\n",
+    "\n",
+    "        polygon = in_polygon(lat, lon)\n",
+    "        \n",
+    "        if polygon and point_list.options:\n",
+    "            if (in_polygon(float(point_list.options[0].split(', ')[0]), float(point_list.options[0].split(', ')[1]))) == polygon:\n",
+    "                    \n",
+    "                clear_marker()\n",
+    "\n",
+    "                latitude.value = f'{lat}'\n",
+    "                longitude.value = f'{lon}'\n",
+    "\n",
+    "                m.add_layer(Marker(location=kwargs.get('coordinates'), name='Marker', draggable=False))\n",
+    "                \n",
+    "        elif polygon and not point_list.options:\n",
+    "                \n",
+    "            clear_marker()\n",
+    "\n",
+    "            latitude.value = f'{lat}'\n",
+    "            longitude.value = f'{lon}'\n",
+    "\n",
+    "            m.add_layer(Marker(location=kwargs.get('coordinates'), name='Marker', draggable=False))\n",
+    "\n",
+    "\n",
+    "def handle_run(arg):\n",
+    "    if not point_list.options:\n",
+    "        print('Please add at least one point before continuing')\n",
+    "        \n",
+    "    elif not start_date.value and not end_date.value:\n",
+    "        print('Please enter a start and end date before continuing')\n",
+    "      \n",
+    "    else:\n",
+    "        print('Continuing...')\n",
```

Comment:
I'm wondering if your code where you start the process, check the status, and obtain the output file should be moved to this function so that it's getting more use. We can see what James says regarding how much this button should do.